src/modules/Elsa.MongoDb/Serializers/TypeSerializer.cs

```diff
@@ -0,0 +1,54 @@
+using MongoDB.Bson;
+using MongoDB.Bson.Serialization;
+
+namespace Elsa.MongoDb.Serializers;
+
+public class TypeSerializer : IBsonSerializer<Type>
+{
+    public Type ValueType => typeof(Type);
+
+    object IBsonSerializer.Deserialize(BsonDeserializationContext context, BsonDeserializationArgs args)
+    {
+        return Deserialize(context, args);
+    }
+
+    public Type Deserialize(BsonDeserializationContext context, BsonDeserializationArgs args)
+    {
+        if (context.Reader.CurrentBsonType == BsonType.Null)
+        {
+            context.Reader.ReadNull();
+            return null;
+        }
+
+        var typeAsString = context.Reader.ReadString();
+        return Type.GetType(typeAsString);
+    }
+
+    public void Serialize(BsonSerializationContext context, BsonSerializationArgs args, Type value)
+    {
+        if (value == null)
+        {
+            context.Writer.WriteNull();
+        }
+        else
+        {
+            context.Writer.WriteString(value.AssemblyQualifiedName);
```

Comment:
Maybe better to use `GetSimpleAssemblyQualifiedName()`, because that excludes version and signature details from the assembly qualified name - which makes it more resilient in case people update packages containing different assembly versions.